Data/Graph/Inductive/Query/Cycles.hs

```diff
@@ -0,0 +1,280 @@
+{- |
+   Module      : Data.Graph.Inductive.Query.Cycles
+   Description : Algorithms for finding all cycles.
+   Copyright   : (c) Gabriel Hjort Blindell 2017
+   License     : 2-Clause BSD
```

Comment:
I decided to include a header, with the BSD3 license and copyright held by both me and you (due to copied functions from Graphalize package).